src/gui/src/components/query-builder/ast-utils.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  selector,
+  root,
+  attribute as createAttribute,
+} from 'postcss-selector-parser'
+import type {
+  tag,
+  id,
+  attribute,
+  combinator,
+  pseudo,
+  string,
+} from 'postcss-selector-parser'
+import type { PostcssNode } from '@vltpkg/dss-parser'
+
+import type { UiNode } from '@/components/query-builder/ui-node-types.ts'
+
+export const toUiNodeFromPostcss = (
+  postcssNode: PostcssNode,
+): UiNode => {
+  const nodeType = postcssNode.type
+  return {
+    type: nodeType as UiNode['type'],
+    value: (postcssNode as { value: string }).value,
+  } as UiNode
+}
+
+export const createSelector = () => selector({ value: '' })
+export const createRoot = () => root({ value: '' })
+
+export type PostCSSLeaf =
+  | ReturnType<typeof tag>
+  | ReturnType<typeof id>
+  | ReturnType<typeof attribute>
+  | ReturnType<typeof combinator>
+  | ReturnType<typeof pseudo>
+  | ReturnType<typeof string>
```

Comment:
it would be nice to have this live in the `@vltpkg/dss-parser` library so that other parts of the code could potentially reuse helpers like this